src/server/svr_jobfunc.c

```diff
@@ -5507,8 +5507,10 @@ svr_setjob_histinfo(job *pjob, histjob_type type)
 	 * through the work task list of the job and delete them using
 	 * delete_task().
 	 */
-	while ((pwt = (struct work_task *)GET_NEXT(pjob->ji_svrtask)) != NULL)
+	while ((pwt = (struct work_task *)GET_NEXT(pjob->ji_svrtask)) != NULL){
+		free(pwt->wt_event2);	/* wt_event2 either has additional data (like msgid) or NULL */
```

Comment:
Why not free this in delete_task() rather than here? The delete_task() function gets called from many places. Since pwt->wt_event2 is initialized to NULL in set_task(), it should be safe. However, you'll need to ensure there are no double free()'s. For example, del_deferred_hook_cmds() also free()'s wt_event2 before it calls delete_task(). Just look for wt_event2 and identify the cases where it gets free()'d. That seems like the more modular way to manage this data.

Reply:
I agree that would have been a better solution. But when I looked at all the places where delete_task() function is used, wt_event2 was not getting freed every time before calling delete_task() function. I think there might be cases where wt_event2 member is needed and is not free()'d intentionally. 
If we put free(pwt->wt_event2) in delete_task() function, then it will be getting free()'d every time, even if we need that structure member later.

Reply:
Yes @sakshamgarg you are right. The data pointed to by some members of tha ptask structure can continue to live even after the ptask itself is freed - case in point are the batch request structures.